libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/speech/InstructionDownloadTask.java

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.services.android.navigation.ui.v5.voice.speech;
+
+import android.os.AsyncTask;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import okhttp3.ResponseBody;
+
+class InstructionDownloadTask extends AsyncTask<ResponseBody, Void, File> {
+  private static int instructionNamingInt = 1;
+  private final String cacheDirectory;
+  private final TaskListener taskListener;
+
+  InstructionDownloadTask(String cacheDirectory, TaskListener taskListener) {
+    this.cacheDirectory = cacheDirectory;
+    this.taskListener = taskListener;
+  }
+
+  @Override
+  protected File doInBackground(ResponseBody... responseBodies) {
+    return saveAsFile(responseBodies[0]);
+  }
+
+  /**
+   * Saves the file returned in the response body as a file in the cache directory
+   *
+   * @param responseBody containing file
+   * @return resulting file, or null if there were any IO exceptions
+   */
+  private File saveAsFile(ResponseBody responseBody) {
+    try {
+      File file = new File(cacheDirectory + File.separator + instructionNamingInt++ + ".mp3");
+      InputStream inputStream = null;
+      OutputStream outputStream = null;
+
+      try {
+        inputStream = responseBody.byteStream();
+        outputStream = new FileOutputStream(file);
+        byte[] buffer = new byte[4096];
+        int numOfBufferedBytes;
+
+        while ((numOfBufferedBytes = inputStream.read(buffer)) != -1) { // -1 denotes end of file
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/pull/751#discussion_r179144355
Sorry I didn't make myself clear. I was referring to the magic number. What about extracting the `-1` magic number into a constant and give it a name based on the comment?
See my previous comment on code comments ☝️